istioctl/cmd/kubeinject.go

```diff
@@ -274,59 +477,15 @@ kube-inject on deployments to get the most up-to-date changes.
 					}
 				}()
 			}
-
-			var meshConfig *meshconfig.MeshConfig
-			if meshConfigFile != "" {
-				if meshConfig, err = mesh.ReadMeshConfig(meshConfigFile); err != nil {
-					return err
-				}
-			} else {
-				if meshConfig, err = getMeshConfigFromConfigMap(kubeconfig, "kube-inject", opts.Revision); err != nil {
-					return err
-				}
-			}
-
-			var sidecarTemplate inject.Templates
-			if injectConfigFile != "" {
-				injectionConfig, err := ioutil.ReadFile(injectConfigFile) // nolint: vetshadow
-				if err != nil {
-					return err
-				}
-				injectConfig, err := readInjectConfigFile(injectionConfig)
-				if err != nil {
-					return multierror.Append(err, fmt.Errorf("loading --injectConfigFile"))
-				}
-				sidecarTemplate = injectConfig
-			} else if sidecarTemplate, err = getInjectConfigFromConfigMap(kubeconfig, opts.Revision); err != nil {
-				return err
-			}
-
 			var valuesConfig string
-			if valuesFile != "" {
-				valuesConfigBytes, err := ioutil.ReadFile(valuesFile) // nolint: vetshadow
-				if err != nil {
-					return err
-				}
-				valuesConfig = string(valuesConfigBytes)
-			} else if valuesConfig, err = getValuesFromConfigMap(kubeconfig, opts.Revision); err != nil {
+			var sidecarTemplate inject.Templates
+			var meshConfig *meshconfig.MeshConfig
+			injector, meshConfig, err := setupKubeInjectParameters(&sidecarTemplate, &valuesConfig, opts.Revision, "kube-inject")
```

Comment:
nit: if kube-inject is the only every value for `command` maybe it doesn't need a parameter?

Reply:
fixed